Package.swift

```diff
@@ -2,23 +2,64 @@
 
 import PackageDescription
 
+let MeiliSearch = "MeiliSearch"
+let MeiliSearchCore = "MeiliSearchCore"
+let MeiliSearchNIO = "MeiliSearchNIO"
+
+var products: [PackageDescription.Product] = [
+  .library(name: MeiliSearch, targets: [MeiliSearch]),
+  .library(name: MeiliSearchCore, targets: [MeiliSearchCore])
+]
+
+var dependencies = [PackageDescription.Package.Dependency]()
+
+var targets: [PackageDescription.Target] = [
+  .target(
+    name: MeiliSearch,
+    dependencies: [
+      .target(name: MeiliSearchCore)
+    ]
+  ),
+  .target(
+    name: MeiliSearchCore, 
```

Comment:
comment: this new target should not break the Podspec as it does a global search for all Swift files within the Sources directory, however when we do add the NIO target it will fail to compile due to the lack of the dependency.

AHC (or NIO) is not available officially on CocoaPods, so this will be a SPM-only feature. I'm okay with that but we must document it and ensure that CocoaPods does continue to work (even if only using Foundation).